.circleci/config.yml

```diff
@@ -498,6 +493,10 @@ commands:
             echo export NPM_CONFIG_PREFIX=/deps/ >> $BASH_ENV
             echo export CC=\"`python -c 'import sysconfig; print(sysconfig.get_config_var("CC"))'`\" >> $BASH_ENV
             cat $BASH_ENV
+      - run:
+          name: Set version
+          command: |
+            make -f Makefile-os version DOCKER_VERSION="$CIRCLE_BRANCH" DOCKER_COMMIT="$CIRCLE_SHA1"
```

Comment:
Here and below in the `image_tag` (and in the Github Action) I wonder if we should use something more unique and also safer (since branches can contain any characters). Maybe `build-${CIRCLE_BUILD_NUM}` or something like that instead ? And try to keep the GitHub Action consistent

Reply:
Can you describe how it should be unique? Like you cannot push the same tag from circle and GHA? Or unique across builds of the same branch? And how safe? What is the thing we need to be safe from?

The GHA and the CircleCI job are consistent in that if you run the build-docker-image job it will tag the branch name, and in circleCI it does the same.

For PRs github uses special detached head branches, AFAIK circleci doesn't really have this. I also think it's somewhat unimportant to have 1:1 parity as the GHA will replace CircleCI soon enough.

Can you clarify so I know more about what specific changes you'd like to see?

Reply:
Regarding uniqueness: contributors re-using a branch name already used for some reason. Including `master` if coming from a fork (not sure if the `CIRCLE_BRANCH` contains the fork name)

Regarding safety: git branch names can contain... any character. So I was worried about potential shell escape issues.

Reply:
secrets don't pass to forks so they cannot push to our docker registry no matter the branch name. But it is a vulnerability for us. I added a `ci-` prefix in the GHA build. Branch builds in Circle never push so I just re-hardcoded the TAG.

Reply:
We also can, and probably should forbid branches "latest" "master" "main" etc.. no reason to allow them to exist. https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule

Reply:
https://github.com/mozilla/addons-server/actions/runs/8848824075/job/24299408303